src/pdn/src/grid.cpp

```diff
@@ -1245,6 +1245,14 @@ ShapeTreeMap InstanceGrid::getInstancePins(odb::dbInst* inst)
     if (net != nullptr) {
       for (auto* mpin : iterm->getMTerm()->getMPins()) {
         for (auto* box : mpin->getGeometry()) {
+          if (box->getTechLayer() == NULL) {
+            getLogger()->warn(utl::PDN,
+                              189,
+                              "no layer for instance {} net {}\n",
```

Comment:
There is no need for a trailing \n in a message, the logger will take care of that.

I think it would be better to put the iterm name in the message rather than the net.  Perhaps

"No layer found for terminal {} of instance{}",  iter->getMTerm->getName(), instance->getName()

Note that you don't need c_str() here as the formatter handles std::string

Reply:
fixed